client/src/components/Tea/TeaList/TeaListContainer.jsx

```diff
@@ -41,20 +46,96 @@ export class TeaListContainer extends React.Component {
 
       const revSortOrder = [...newSortOrder].reverse();
 
-      currentState = {
+      currentSortState = {
         columnName: columnName,
         asc: newSortOrder,
         desc: revSortOrder
       };
     }
     this.setState({
       sortedIDs: newSortOrder,
-      memoizedIDs: currentState,
+      memoizedIDs: currentSortState,
       sortColumn: columnName,
       sortOrder: sortOrder
     });
   };
 
+  filterChangeHandler = event => {
+    const name = event.target.name;
+    const value = event.target.value;
+
+    this.setState({
+      ...this.state.formControls,
+      formControls: {
+        ...this.state.formControls,
+        [name]: value
+      }
+    });
+  };
+
+  handleFilterClick = event => {
+    event.preventDefault();
+    const list = this.state.sortedIDs;
+    let currentFilterState = this.state.memoizedIDs;
+    let newFilterOrder;
+
+    if (
+      this.state.formControls.filterCategory !== "" &&
+      this.state.formControls.filterCriteria !== ""
+    ) {
+      newFilterOrder = list.filter(item => {
+        // this returns NaN to get a falsy value if the Number function doesn't return an integer
+        if (Number(this.state.formControls.filterCriteria)) {
+          // I am actively leveraging the coercion of the double equals here
+          return (
+            this.props.teas.allTeas[item][
+              this.state.formControls.filterCategory
+            ] == this.state.formControls.filterCriteria
+          );
```

Comment:
I don't understand this code. Why is `filterCriteria` stored as either a number or a string?

Reply:
the input value comes over as a string, but the value in the redux store/coming from the db is an integer.